ADR/0019-customize-url-github.md

```diff
@@ -0,0 +1,152 @@
+there is no bookmarkable URL which allows users to see the logs of a specific TaskRun. This feature
+may be added in the near future (see [HAC-3370](https://issues.redhat.com/browse/HAC-3307)).
+
+The following components in AppStudio send data/results back to GitHub via the Checks API:
+
+- Pipelines as Code (PaC)
+- Integration Service
+- Other services which may run Pipelines in response to a GitHub event (ex - potentially Release Service).
+
+An AppStudio workspace is associated with a Kubernetes namespace on a AppStudio “member cluster”,
+which is not directly accessible by end users. The namespace name on the member cluster currently
+correlates with the workspace name (a suffix is added). [ADR-0010](./0010-namespace-metadata.md)
+specifies that a namespace label should exist which links the member cluster namespace back to the
+AppStudio workspace.
+
+Artifacts in AppStudio are organized into Applications and Components. A namespace can have more
+than one `Application`, and an `Application` can have one or more `Components`. At present, each
+`Component` has an associated PaC `Repository` object, though this may be subject to change in the future.
+
+### Requirements
+
+For a given PipelineRun in AppStudio, the following information must be identifiable based on
+information in the AppStudio member cluster:
+
+- Workspace ID
+- Application ID
+- Pipeline run ID
+
+AppStudio components that report data back to GitHub must know the root URL for the AppStudio UI.
+This URL must be configurable per AppStudio deployment (ex - staging, production in 
+[infra-deployments](https://github.com/redhat-appstudio/infra-deployments)).
+Workspace and application IDs should _not_ be added to PipelineRun YAML that is added to git
+repositories for PaC. PaC PipelineRun templates should be transportable when a user forks someone
+else’s git repo.
+
+## Decision
+
+AppStudio components that provide `PipelineRun` result links to GitHub or other source control
+management systems must provide URLs that are accessible to registered end users who have
+permission to view the requested `PipelineRun` resource.
+
+## Consequences
+
+### Dev Sandbox: Member Cluster Labels
+
+Member cluster namespaces MUST have the `appstudio.redhat.com/workspace_name` label, in accordance
+with [ADR-0010](./0010-namespace-metadata.md).
+
+### Pipelines as Code: Customize URLs per Repository
+
+Pipelines as Code's `Repository` custom resource will be extended to let users/service providers
+customize the `PipelineRun` URL sent to SCM providers. This will take advantage of the upcoming
+parameters feature in Pipelines as Code (see [SRVKP-2940](https://issues.redhat.com/browse/SRVKP-2940)).
+
+Below is a proposed API, with an implementation for AppStudio:
+
+```yaml
+spec:
+  params:
+    - name: workspace
+      type: string
+      value: adkaplan-demo
+    - name: application
+      type: string
+      value: demo-app
+  customConsole:
```

Comment:
as discussed on slack we don't need a specific customConsole.

params will be implemented on Repository CR (PR and Doc here https://github.com/openshift-pipelines/pipelines-as-code/pull/1186).

The next step will be to plug those custom params onto the custom-console global config.  (we don't have a SRVKP for that).

This means.

If in the custom-console global config you have: 

```
custom-console-url: https://{{ application }}/pr/{{ workspace }}/logs
```

it willl grab the value from the params as defined on the Repo CR of the user and expand it on the url.

We probably going to have a custom-console-url-fallback global setting for a fallback URL if for whatever reasons those params were not populated...

we don't do ADR or such for OSP but happy to talk about it on this PR if this implementation is not fulfilling your reqs.

Reply:
> The next step will be to plug those custom params onto the custom-console global config. (we don't have a SRVKP for that).

I think https://issues.redhat.com/browse/SRVKP-2944 covers this feature?

Reply:
oh yeah thanks, getting old and forgetting stuff i did just last week 👴🏻